### ISSUE TITLE
dep key items unintended merging

### ISSUE THREAD

**Issue by majidaldo** [ESSS/conda-devenv#152]
```
#devenv file
name: 'sdf'
dependencies:
  - d1
  - d2
  - notdevenvkey:
    - one
    - two
    - startwith xxx
    - startwith yyy
 ```
gets processed to `conda devenv --print-full`
```
# generated by conda-devenv, do not modify and do not commit to VCS
dependencies:
- d1
- d2
- notdevenvkey:
  - one
  - startwith xxx,yyy
  - two
name: sdf
environment: {}
```
i'm not sure if this behavior is by design.

**Comment by prusse-martin** [comment]
It recursively merge the dicts, that is by design. It has some custom handling for "pip" and "name" though.
Thank you the small example to make clear to issue.
Does this present a problem for your use case?

**Comment by majidaldo** [comment]
> It recursively merge the dicts, that is by design. It has some custom handling for "pip" and "name" though. Thank you the small example to make clear to issue. Does this present a problem for your use case?

yes. i slightly post-process the conda-devenv output. i think conda-devenv should only process what's going to conda and leave the rest as is.

**Comment by majidaldo** [comment]
> > It recursively merge the dicts, that is by design. It has some custom handling for "pip" and "name" though. Thank you the small example to make clear to issue. Does this present a problem for your use case?
> 
> yes. i slightly post-process the conda-devenv output. i think conda-devenv should only process what's going to conda and leave the rest as is.

for example you can think about installers other than conda or pip like npm. each installer has a spec format that shouldn't be touched by conda-devenv.

**Comment by prusse-martin** [comment]
The idea was to also merge the specs for other installers.
The custom pip handling affect just some declarations not all.

**Comment by majidaldo** [comment]
> 

why would it? each installer has its own way. also, the specs got alphabetically ordered. 

related https://github.com/ESSS/conda-devenv/issues/84
